### ISSUE TITLE
Add an option to simply pass AwilixContainer via the plugin options 

### ISSUE THREAD

**Issue by yshrsmz** [fastify/fastify-awilix#124]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

I want to propose an option to pass a custom AwilixContainer to the plugin.

### Motivation

Hi, thanks for the great plugin.

Currently, this plugin uses a singleton container declared inside it.
While we can directly access it via `import { diContainer } from '@fastify/awilix'` and call `diContainer.dispose()` to reset its contents, I think it's more intuitive if we can just pass a new AwilixContainer when we initialize the plugin.

This proposal is mainly for testing. Sure, we can dispose and re-initialize deps on-the-fly with the method described above, but the option to provide the diContainer makes the test far easier.

### Example

```ts
export type FastifyAwilixOptions = {
  disposeOnResponse?: boolean
  disposeOnClose?: boolean
  asyncInit?: boolean
  asyncDispose?: boolean
  eagerInject?: boolean
  container?: AwilixContainer<Cradle> // <- default to `diContainer` defined in the plugin
}

// register AwilixContainer with plugin option.
// this way we can easily swap the dependencies in testing environment.
fastify.register(fastifyAwilixPlugin, {
  container: newDIContainer
})
```

**Comment by yshrsmz** [comment]
(and with this implementation, we can merge plugin code for each injectionMode)

**Comment by kibertoad** [comment]
Sure, I'll look into it!

**Comment by kibertoad** [comment]
@yshrsmz It was released in 5.0.0, could you please test it and report back if it works as expected?
